doc/explanation/api/reactivity.md

```diff
@@ -0,0 +1,177 @@
+
+## Reactivity in Panel
+
+Now that we have introduced some of the concepts behind the reactive approaches let's dive into the nitty gritty and try to understand how Panel actually implements the abstract concepts of reactivity in practice. As we discussed previously, reactive approaches rely on **data binding** and we also mentioned that Parameters are the native way for Panel to express the data model. At the implementation level data binding therefore happens by binding parameter state to some component or function. The important thing to understand is that for binding to be expressed in code we have to make a distinction between the parameter **value** and the abstract parameter **reference**. To make this concrete, to perform binding we have to use the parameter **object**, which acts as a proxy or reference for its value. Let's say we have a `TextInput` widget:
+
+```python
+text = pn.widgets.TextInput(value='Hello world!')
+```
+
+The widget's value parameter value can be accessed with `text.value`, but this represents the value at that point in time. To perform binding we have to access the parameter object with `text.param.value`, this acts as a reference or proxy that can be bound. In Panel there are two main ways of performing data binding, either we can directly bind a reference to a parameter, e.g. we can do:
+
+```python
+pn.pane.Markdown(object=text.param.value)
+```
+
+This performs data binding between the `value` parameter of the widget and the `object` parameter of the `Markdown` pane and provides a declarative specification of this reactive component. We will refer to this as **component level binding**. This is the simplest and most explicit form of data binding in Panel and has to be distinguished from **function level binding** where we use `pn.bind` to bind a parameter reference to a function:
+
+```python
+a_slider = pn.widgets.FloatSlider(start=0, end=10)
+b_slider = pn.widgets.FloatSlider(start=0, end=10)
+
+def add(a, b):
+    return pn.pane.Str(f'{a} + {b} = {a+b}')
+
+pn.Column(a_slider, b_slider, pn.bind(add, a_slider.param.value, b_slider.param.value))
+```
+
+In this example, we have bound two inputs to the function and return an output, the `Str` pane. Internally a few things will happen to make this work:
+
+1. Panel wraps the bound function in a so called `ParamFunction` pane.
+2. The `ParamFunction` pane watches the inputs to the bound function for changes.
+3. When a slider is dragged and the input changes the `ParamFunction` will re-evaluate the function with the new inputs and then figure out the appropriate update.
+
+:::{important}
+Since Panel cannot know if you have a handle on the `Str` pane that is being returned, it cannot safely update the `Str` pane inplace. Therefore it re-renders the corresponding model every time the inputs change. For complex panes and output this can lead to undesirable flicker. To avoid this you have to tell Panel that the output can be safely updated `inplace`:
```

Comment:
I don't understand "Since Panel cannot know if you have a handle on the `Str` pane that is being returned, it cannot safely update the `Str` pane inplace.".